### ISSUE TITLE
crash with watching on -- html/template internal error: template escaping out of sync 

### ISSUE THREAD

**Issue by pbnjay** [spf13/hugo#1405]
Just saw an odd crash while editing a post with watching enabled. I'm editing in Sublime Text, using the SFTP plugin to sync on save to my server where hugo is running. After one of my saves I got this crash, my best guess is that the file wasn't completely synced yet? Although I'm editing a markdown content file, so I'm not sure why that is affecting the template state.

I have not been able to reproduce this one with the exact same files that were in place, so it seems like it may be a race of some sort.

I'm running on the latest binary package, invocation was `hugo server -t hugo-incorporated -w -p 3000 -b http://example.com --appendPort=false`

Snip of running goroutine, I can post the full crash dump if interested:
```
Change detected, rebuilding site 
2015-09-05 18:41 +0000 
panic: html/template internal error: template escaping out of sync [recovered] 
        panic: html/template internal error: template escaping out of sync 
 
goroutine 820 [running]: 
text/template.errRecover(0xc208263a50) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:104 +0x14f 
html/template.(*Template).lookupAndEscapeTemplate(0xc208124580, 0xc208281b00, 0x1b, 0xc2082dea50, 0x0, 0x0) 
        /usr/local/Cellar/go/1.4.2/libexec/src/html/template/template.go:112 +0x3a6 
html/template.(*Template).ExecuteTemplate(0xc208124580, 0x7f4823e316b8, 0xc2080f7ab0, 0xc208281b00, 0x1b, 0xae1020, 0xc20803ac00, 0x0, 0x0) 
        /usr/local/Cellar/go/1.4.2/libexec/src/html/template/template.go:88 +0x54 
github.com/spf13/hugo/tpl.ExecuteTemplate(0xae1020, 0xc20803ac00, 0xc2080f7ab0, 0xc208281ac0, 0x2, 0x2) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/tpl/template.go:114 +0x1d6 
github.com/spf13/hugo/tpl.ExecuteTemplateToHTML(0xae1020, 0xc20803ac00, 0xc208281ac0, 0x2, 0x2, 0x0, 0x0) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/tpl/template.go:131 +0xaa 
github.com/spf13/hugo/tpl.Partial(0xc2082e3041, 0xc, 0xc208148e90, 0x1, 0x1, 0x0, 0x0) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/tpl/template.go:100 +0x260 
reflect.Value.call(0x9c8640, 0xc4bef0, 0x13, 0xaec750, 0x4, 0xc20820c780, 0x2, 0x2, 0x0, 0x0, ...) 
        /usr/local/Cellar/go/1.4.2/libexec/src/reflect/value.go:419 +0x10e5 
reflect.Value.Call(0x9c8640, 0xc4bef0, 0x13, 0xc20820c750, 0x2, 0x2, 0x0, 0x0, 0x0) 
        /usr/local/Cellar/go/1.4.2/libexec/src/reflect/value.go:296 +0xbc 
text/template.(*state).evalCall(0xc2081ce640, 0xae1020, 0xc20803ac00, 0x16, 0x9c8640, 0xc4bef0, 0x13, 0x7f4823e3c948, 0xc2080ffd70, 0xc2082e3038, ...) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:567 +0xaa4 
text/template.(*state).evalFunction(0xc2081ce640, 0xae1020, 0xc20803ac00, 0x16, 0xc2080ffda0, 0x7f4823e3c948, 0xc2080ffd70, 0xc208235e80, 0x3, 0x4, ...) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:459 +0x33b 
text/template.(*state).evalCommand(0xc2081ce640, 0xae1020, 0xc20803ac00, 0x16, 0xc2080ffd70, 0x0, 0x0, 0x0, 0x0, 0x0, ...) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:359 +0x210 
text/template.(*state).evalPipeline(0xc2081ce640, 0xae1020, 0xc20803ac00, 0x16, 0xc208038be0, 0x0, 0x0, 0x0) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:332 +0x1a8 
text/template.(*state).walk(0xc2081ce640, 0xae1020, 0xc20803ac00, 0x16, 0x7f4823e31408, 0xc2080ffe60) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:167 +0x145 
text/template.(*state).walk(0xc2081ce640, 0xae1020, 0xc20803ac00, 0x16, 0x7f4823e3c8b8, 0xc2080fb980) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:175 +0x862 
text/template.(*Template).Execute(0xc208235680, 0x7f4823e316b8, 0xc208011500, 0xae1020, 0xc20803ac00, 0x0, 0x0) 
        /usr/local/Cellar/go/1.4.2/libexec/src/text/template/exec.go:155 +0x3f2 
html/template.(*Template).ExecuteTemplate(0xc208124580, 0x7f4823e316b8, 0xc208011500, 0xc2082801c0, 0x1a, 0xae1020, 0xc20803ac00, 0x0, 0x0) 
        /usr/local/Cellar/go/1.4.2/libexec/src/html/template/template.go:92 +0xc0 
github.com/spf13/hugo/hugolib.(*Site).renderThing(0xc2083161c0, 0xae1020, 0xc20803ac00, 0xc2082801c0, 0x1a, 0x7f4823e316b8, 0xc208011500, 0x0, 0x0) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/site.go:1500 +0x210 
github.com/spf13/hugo/hugolib.(*Site).render(0xc2083161c0, 0xc2081ce580, 0x3a, 0xae1020, 0xc20803ac00, 0xc208011500, 0xc20813c000, 0xa, 0x10, 0x0, ...) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/site.go:1475 +0x2bd 
github.com/spf13/hugo/hugolib.(*Site).renderAndWritePage(0xc2083161c0, 0xc2081ce580, 0x3a, 0xc2081ce4c0, 0x32, 0xae1020, 0xc20803ac00, 0xc20813c000, 0xa, 0x10, ...) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/site.go:1426 +0xdf 
github.com/spf13/hugo/hugolib.pageRenderer(0xc2083161c0, 0xc208066ba0, 0xc208066ae0, 0xc2082e0080) 
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/site.go:965 +0x47e 
created by github.com/spf13/hugo/hugolib.(*Site).RenderPages 
        /Users/spf13/gopath/src/github.com/spf13/hugo/hugolib/site.go:923 +0x124 
```

**Comment by bep** [comment]
Thanks for the report.

This duplicates #1326 and #917. 

This data race is a bug in Go's stdlib and is fixed in Go 1.5 -- and as such will be fixed in Hugo 0.15.

**Comment by pbnjay** [comment]
Great thanks for the update, for some reason those didn't come up in my search. Sorry for the dupe!
